<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="David Jordan">
<meta name="author" content="Mihoko Tame">
<meta name="dcterms.date" content="2025-02-28">

<title>Recreating the Gal-His yeast system used to demonstrate adaptive reprogramming – Notebooks</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark-d166b450ba5a8e9f7a0ab969bf6592c1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap-22381ab97ffb8a420d3841344730e94d.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top quarto-banner">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Notebooks</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../index.html"> 
<span class="menu-text">Notes Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../tech.html"> 
<span class="menu-text">Technical Notes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../builds.html"> 
<span class="menu-text">Build Notes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../results.html"> 
<span class="menu-text">Results Notes</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://livingphysics.org"> <i class="bi bi-globe" role="img">
</i> 
<span class="menu-text">Main Site</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../index.xml"> <i class="bi bi-rss" role="img">
</i> 
<span class="menu-text">RSS Feed</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Recreating the Gal-His yeast system used to demonstrate adaptive reprogramming</h1>
                                <div class="quarto-categories">
                <div class="quarto-category">Builds</div>
                <div class="quarto-category">Notes</div>
                <div class="quarto-category">Yeast</div>
              </div>
                  </div>
  </div>
    
  
  <div class="quarto-title-meta">

      <div>
      <div class="quarto-title-meta-heading">Authors</div>
      <div class="quarto-title-meta-contents">
               <p><a href="mailto:public@livingphysics.org">David Jordan</a> </p>
               <p>Mihoko Tame </p>
            </div>
    </div>
      
      <div>
      <div class="quarto-title-meta-heading">Published</div>
      <div class="quarto-title-meta-contents">
        <p class="date">February 28, 2025</p>
      </div>
    </div>
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#overview" id="toc-overview" class="nav-link active" data-scroll-target="#overview">Overview</a>
  <ul class="collapse">
  <li><a href="#plasmid-reconstruction" id="toc-plasmid-reconstruction" class="nav-link" data-scroll-target="#plasmid-reconstruction">Plasmid Reconstruction</a>
  <ul class="collapse">
  <li><a href="#cloning" id="toc-cloning" class="nav-link" data-scroll-target="#cloning">Cloning</a></li>
  </ul></li>
  <li><a href="#yeast-strains" id="toc-yeast-strains" class="nav-link" data-scroll-target="#yeast-strains">Yeast Strains</a></li>
  <li><a href="#transformation" id="toc-transformation" class="nav-link" data-scroll-target="#transformation">Transformation</a></li>
  <li><a href="#transition-to-galactose" id="toc-transition-to-galactose" class="nav-link" data-scroll-target="#transition-to-galactose">Transition to galactose</a></li>
  </ul></li>
  <li><a href="#procedures" id="toc-procedures" class="nav-link" data-scroll-target="#procedures">Procedures</a>
  <ul class="collapse">
  <li><a href="#cloning-1" id="toc-cloning-1" class="nav-link" data-scroll-target="#cloning-1">Cloning</a>
  <ul class="collapse">
  <li><a href="#generating-the-dna-inserts" id="toc-generating-the-dna-inserts" class="nav-link" data-scroll-target="#generating-the-dna-inserts">Generating the DNA inserts</a></li>
  <li><a href="#plasmids" id="toc-plasmids" class="nav-link" data-scroll-target="#plasmids">Plasmids</a></li>
  <li><a href="#restriction-enzymes" id="toc-restriction-enzymes" class="nav-link" data-scroll-target="#restriction-enzymes">Restriction Enzymes</a></li>
  <li><a href="#restriction-pcr" id="toc-restriction-pcr" class="nav-link" data-scroll-target="#restriction-pcr">Restriction PCR</a></li>
  <li><a href="#yeast-liquid-culture" id="toc-yeast-liquid-culture" class="nav-link" data-scroll-target="#yeast-liquid-culture">Yeast Liquid Culture</a></li>
  <li><a href="#yeast-plating" id="toc-yeast-plating" class="nav-link" data-scroll-target="#yeast-plating">Yeast Plating</a></li>
  <li><a href="#yeast-strains-1" id="toc-yeast-strains-1" class="nav-link" data-scroll-target="#yeast-strains-1">Yeast Strains</a></li>
  <li><a href="#transformation-1" id="toc-transformation-1" class="nav-link" data-scroll-target="#transformation-1">Transformation</a></li>
  <li><a href="#selectable-markers" id="toc-selectable-markers" class="nav-link" data-scroll-target="#selectable-markers">Selectable Markers</a></li>
  </ul></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">





<blockquote class="blockquote">
<p>A goal for the future would be to determine the extent of knowledge the cell has of itself and how it uses that knowledge in a thoughtful manner when challenged.</p>
<p>(McClintock 1984)</p>
</blockquote>
<p>This note describes how I am recreating the strain from the Braun lab <span class="citation" data-cites="Stolovicki2006-sd">&nbsp;[<a href="#ref-Stolovicki2006-sd" role="doc-biblioref">1</a>]</span> used to demonstrate adaptive reprogramming to novel challenges in the yeast <em>Saccharomyces cerevisiae</em> Unfortunately the strain and plasmid were lost due to a freezer fault. This document is written informally but should be understandable to a reasonably interested secondary school student. For brevity, the main document uses standard scientific terminology, but these terms are explained in detail in the <a href="#procedures">Procedures</a> section.</p>
<section id="overview" class="level1">
<h1>Overview</h1>
<p>In the context of computing, one might think of a single celled organism like a very complex “look-up table”, that is a passive input output device that has been optimized by evolution over very long time scales to give an “optimal” response to an environmental input. In single cell metabolism, inspired by the early work of Monod, this point of view has been very powerful for understanding microbial metabolic regulation. However, the concept of cell learning proposes that cells are not simply look up tables, but are capable of more sophisticated computation. To make this concrete, let’s look at a simplified, idealized “circuit” involved in sugar utilisation. Yeast prefer to use glucose as a carbon source rather than galactose. Prefer here means a few things: First empirically, yeast grow about 50% slower on galactose (~0.4 doublings/hr) than on glucose (~0.2 doublings/hr). Second, yeast have a gene regulatory system called the “catabolite repression system” that turns off the GAL regulon (all the genes required to uptake and utilize galactose) in the presence of glucose. Interestingly, galactose does not seem to be a less efficient source of ATP, but the increased number of enzymes required to bring it into the glycolysis pathway compared to glucose (4 vs.&nbsp;1) seems to have made it the less preferred sugar. With this in mind, we can write down the input-output function for the two presence or absence of the two sugars (inputs) and the activity of the Gal promoter (output).</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Glucose</th>
<th>Galactose</th>
<th>Gal Promoter Activity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>0</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="even">
<td>1</td>
<td>0</td>
<td>0</td>
</tr>
<tr class="odd">
<td>1</td>
<td>1</td>
<td>0</td>
</tr>
<tr class="even">
<td>0</td>
<td>1</td>
<td>1</td>
</tr>
</tbody>
</table>
<p>Generating this behavior with combinatorial logic could be done as e.g.&nbsp;output = (NOT glucose) AND galactose.</p>
<p>With this view of a cell as a simple evolutionarily optimized input output function, we can ask what would happen if we take this regulatory system and tie its control to a different metabolic output, in this case, to the expression of an enzyme needed to make the amino acid histidine in a yeast that has no other way to make histidine. Then the yeast would face a challenge when grown in glucose, shutting down the promoter means shutting of is only source of histidine. The inherent logic would turn off the Gal promoter but the promoter being on is now required for the yeast to make histidine (and thus to continue to grow). This engineered strain is known as the “Gal-His system”. It was invented by the Braun lab <span class="citation" data-cites="Stolovicki2006-sd">&nbsp;[<a href="#ref-Stolovicki2006-sd" role="doc-biblioref">1</a>]</span>. The traditional view is that genetic changes would be required to rewire the circuit and change the truth table, and that any strain that found the correct mutations to do this would grow and outcompete the others. However, what Braun and colleagues found was that yeast managed to find new steady states of gene expression that allowed the cells to overcome this challenge without mutations, even in single cells that were not dividing <span class="citation" data-cites="woronoffMetabolicCostRapid2020">&nbsp;[<a href="#ref-woronoffMetabolicCostRapid2020" role="doc-biblioref">2</a>]</span>. They called this “adaptive reprogramming”. This note describes recreating the “Gal-His System” as a tool to further investigate this phenomenon and probe the mechanisms and limits of the computational capacity of single cells.</p>
<section id="plasmid-reconstruction" class="level2">
<h2 class="anchored" data-anchor-id="plasmid-reconstruction">Plasmid Reconstruction</h2>
<p>The original plasmid was generated by <a href="#cloning">cloning</a> the His3 gene and a GFP gene into the dual expression plasmid <a href="https://www.agilent.com/en/product/protein-expression/protein-expression-vectors-kits/yeast-expression-vectors/pesc-yeast-epitope-tagging-vectors-232966">pESC-Leu</a> from Agilent. We will use the GFP originally used, a yeast optimized GFP called S65T, but one could use any mutationally optimized GFP originally introduced in <span class="citation" data-cites="cormackYeastenhancedGreenFluorescent1997">&nbsp;[<a href="#ref-cormackYeastenhancedGreenFluorescent1997" role="doc-biblioref">3</a>]</span>. Further information on optimized GFPs can be found in <span class="citation" data-cites="cormackFACSoptimizedMutantsGreen1996">&nbsp;[<a href="#ref-cormackFACSoptimizedMutantsGreen1996" role="doc-biblioref">4</a>]</span>, and a comparison of yeast optimized GFPs can be found in <span class="citation" data-cites="kaishimaExpressionVariedGFPs2016">&nbsp;[<a href="#ref-kaishimaExpressionVariedGFPs2016" role="doc-biblioref">5</a>]</span>. The plasmid has a Leucine selectable marker and a <span class="math inline">\(2\mu\)</span> origin for yeast a and AmpR selective marker and <span class="math inline">\(ori\)</span> origin for growth in <em>E. coli</em>. The plasmid contains a bidirectional Gal1/Gal10 promoter and each promoter has a multiple cloning site (MCS).</p>
<section id="cloning" class="level3">
<h3 class="anchored" data-anchor-id="cloning">Cloning</h3>
<p>We have used the GFP (S65T) cloned from the <a href="https://www.addgene.org/20409/">pKEN GFP mut2</a> plasmid. The HIS3 gene was cloned from the plasmid <a href="https://www.snapgene.com/plasmids/yeast_plasmids/pRS313">pRS33</a>. MT designed forward and reverse PCR primers to <a href="#restriction-pcr">introduce restriction sites</a> for NotI and BglII into the amplified GFP sequence and ApaI and XhoI in the HIS3 gene. The GFP sequence was cloned into the pESC-Leu Gal10 MCS and the HIS3 into the Gal1 MCS in the bi-directional Gal1/Gal10 promoter. Cloning results were confirmed by whole plasmid sequencing.</p>
</section>
</section>
<section id="yeast-strains" class="level2">
<h2 class="anchored" data-anchor-id="yeast-strains">Yeast Strains</h2>
<p>It would be very simple to use store bought yeast for these experiments, and I plan to do that for simpler experiments where no genetic engineering is necessary, however, laboratory strains have many features that make genetic engineering muh easier, such as gene deletions that can be used as <a href="#selectable-markers">selectable markers</a>. For these experiments, we need a yeast strain that cannot make its own Leucine, so that we can use the Lu selectable marker on the plasmid, as well as being unable to make histidine, so that Gal-His construct will be its sole source of histidine during adaptation trials.</p>
<p>The original experiments were done in a yeast strain designated as YPH499. Interestingly, in the supplemental information of this paper by <span class="citation" data-cites="bennettMetabolicGeneRegulation2008">&nbsp;[<a href="#ref-bennettMetabolicGeneRegulation2008" role="doc-biblioref">6</a>]</span>, the authors note that this strain has “significantly impaired galactose uptake”. This strain is derived from S288C, which has a known defect in the GAL2 galactose permease gene (a transporter that brings galactose into the cell). The <em>GAL2</em> gene was supposed to have been repaired in YPH499 but the authors of the above note that there remain a number of point mutations in the coding sequence of the Gal2 galactose transporter compared to the strain K699. They provide evidence that galactose uptake is impaired in the YPH499 strain. We are using the W303 strain, which seems to have the same Gal2 sequence as the K699 stain save one residue at 369.</p>
<blockquote class="blockquote">
<p>Our sequencing results revealed nine point mutations in the YPH499 <em>GAL2</em> sequence, yielding the following five amino acid mutations: V8M P50S S90G Y369S R392H.</p>
</blockquote>
<p>They then use a fluorescent fusion GAL2p and show that is distribution is not uniform on the cell membrane in YPH499 as it is in another strain K699 which has teh <em>wild-type</em> <em>GAL</em> sequence. The strains I have easily available are the W303 strain and the BY4743 strains. The W303 strain is only 85.4% congenic with S288c (and thus with YPH499)</p>
<p>For Gal2, W303 is closer to K699 than to YPH499. For the residues noted, K699 has the residues [VPSYR], YPH499 has the residues [MSGSH] and our strain W303 has [VPSSR] according to publicly available <a href="https://www.yeastgenome.org/variant-viewer#/S000004071">sequencing data</a>.</p>
</section>
<section id="transformation" class="level2">
<h2 class="anchored" data-anchor-id="transformation">Transformation</h2>
<p>The completed pESC-Leu_Gal-His plasmid was transfected into competent yeast cells using the LiAc procedure <span class="citation" data-cites="itoTransformationIntactYeast1983">&nbsp;[<a href="#ref-itoTransformationIntactYeast1983" role="doc-biblioref">7</a>]</span>. We followed the protocols given in <span class="citation" data-cites="gietzFrozenCompetentYeast2007">&nbsp;[<a href="#ref-gietzFrozenCompetentYeast2007" role="doc-biblioref">8</a>]</span> for making competent cells, and then transformed them following the high-efficiency procedure in <span class="citation" data-cites="gietzHighefficiencyYeastTransformation2007">&nbsp;[<a href="#ref-gietzHighefficiencyYeastTransformation2007" role="doc-biblioref">9</a>]</span>. The construct contains a Leucine selectable marker under the control of the native Leu promoter, and the <a href="#w303">W303</a> strain carries the <em>leu2-3</em> allele, therefore only cells successfully transformed will grow on minimal media plates without leucine. Directly after transformation, yeast were grown in recovery media and then plated on selection plates lacking leucine and containing glucose. A mock transformation was also carried out which did not contain any of the plasmid. The results of plating different volumes of the transformation mixture <span class="math inline">\((2\mu l, 20\mu l, 200\mu l )\)</span> are shown for both the control (<span class="math inline">\(\emptyset\)</span>) and the plasmid (Tx) after 3 days of growth at 30C.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/transformation-leu.jpeg" class="img-fluid figure-img"></p>
<figcaption>Transformation results Leu- selection plates</figcaption>
</figure>
</div>
</section>
<section id="transition-to-galactose" class="level2">
<h2 class="anchored" data-anchor-id="transition-to-galactose">Transition to galactose</h2>
<p>After confirming that the plasmid was successfully transformed, the next step was to turn on the Gal1/10 promoter on the plasmid to ensure that both the HIS3 and GFP genes were cloned properly. To do this, cells were first grown for 3 days in minimal media without leucine and galactose as the sugar. After this culture grew to high OD, these cells were plated on to selection plates with galactose and without histidine. After 3 days, many colonies were seen on these plates, indicating that the HIS3 gene was working. Cells plated on glucose plates without histidine did not show any colonies. Finally fluorescence micrographs were take to ensure that the GFP was being expressed.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/Project_Image002_ch00.jpg" class="img-fluid figure-img"></p>
<figcaption>GFP positive yeast, 40x magnification</figcaption>
</figure>
</div>
</section>
</section>
<section id="procedures" class="level1">
<h1>Procedures</h1>
<section id="cloning-1" class="level2">
<h2 class="anchored" data-anchor-id="cloning-1">Cloning</h2>
<section id="generating-the-dna-inserts" class="level3">
<h3 class="anchored" data-anchor-id="generating-the-dna-inserts">Generating the DNA inserts</h3>
<p>This usually involves either cutting the DNA out of an existing source using restriction enzymes<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a>, having the DNA synthesized, or most commonly, amplifying it out of an existing source using the Polymerase Chain Reaction (PCR).</p>
<section id="pcr-amplification" class="level4">
<h4 class="anchored" data-anchor-id="pcr-amplification">PCR amplification</h4>
<p>Polymerase chain reaction, or <a href="https://en.wikipedia.org/wiki/Polymerase_chain_reaction">PCR</a> amplification is a procedure that uses a heat stable DNA polymerase from a thermophilic bacterium called <strong>Thermus aquaticus</strong> to make copies of a given sequence. To do this, special primers are designed that serve as a starting point for the polymerase to add nucleotides. These primers are designed to bind to the two single srtands of the DNA after it is denatured at high temperature. The Polymerase then extends both primers to the end of the fragment, yielding two new double stranded molecules. This procedure is repeated for 20-30 cycles to greatly amplify the number of molecules.</p>
</section>
</section>
<section id="plasmids" class="level3">
<h3 class="anchored" data-anchor-id="plasmids">Plasmids</h3>
<p>Plasmids are circles of DNA containing a sequence that tells a certain cell type to replicate it. These are known and origins of replication. Often plasmids have origins that alow them to be replicated in bacteria, as transforming bacteria and growing them is convenient way to <em>amplify</em> the plasmid. For example, the <a href="https://www.snapgene.com/plasmids/yeast_plasmids/pESC-LEU">pESC-Leu</a> plasmid that we hav used as a backbone for this work contains three origins shown in yellow in the the plasmid map. The bacterial double stranded DNA origin is denote <em>ori</em>. The <em>f1 ori</em> is a phage derived origin that allows for single stranded replication and packaging into a phage. Both of these operate in bacteria. The <span class="math inline">\(2\mu\)</span> origin is the yeast replication origin. The identity of the origin control for example the plasmid copy number in the cell. In addition, cells usually cannot maintain reliably different plasmids with the same origin.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./images/pESC-LEU.png" class="img-fluid figure-img"></p>
<figcaption>pESC-Leu PLasmid Map (from SnapGene)</figcaption>
</figure>
</div>
<p>For more information about plasmid origins of replication, see <a href="https://blog.addgene.org/plasmid-101-origin-of-replication">this Addgene article</a></p>
</section>
<section id="restriction-enzymes" class="level3">
<h3 class="anchored" data-anchor-id="restriction-enzymes">Restriction Enzymes</h3>
<p>Restriction enzymes are proteins that recognize a specific sequence in DNA and cleave the DNA in a predictable manner. Restriction enzymes can be purchased from <a href="https://www.neb.com/en-gb/products/restriction-endonucleases">New England Biolabs</a>.</p>
</section>
<section id="restriction-pcr" class="level3">
<h3 class="anchored" data-anchor-id="restriction-pcr">Restriction PCR</h3>
<p>Restriction sites can be introduced into PCR amplified inserts by designing primers which have a complementary binding region to the fragment to be amplified, and a non complementary region that contains the restriction enzyme recognition site. Usually, a few extra bases are added on the other side of the enzyme recognition site as well to make the enzyme more efficient at cutting the fragment. Below is a table of the PCR amplification primers used. Within the complementary region, the start codon ATG is indicated in uppercase.</p>
<table class="caption-top table">
<colgroup>
<col style="width: 28%">
<col style="width: 17%">
<col style="width: 14%">
<col style="width: 40%">
</colgroup>
<thead>
<tr class="header">
<th>Name</th>
<th style="text-align: left;">Extra Bases</th>
<th style="text-align: left;">RE Site</th>
<th style="text-align: left;">Complementary region</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>GFPs65t-NotI-FW</td>
<td style="text-align: left;">aaaacc</td>
<td style="text-align: left;">gcggccgc</td>
<td style="text-align: left;">atacatATGagtaaaggagaagaac</td>
</tr>
<tr class="even">
<td>GFPs65t-BglII-RV</td>
<td style="text-align: left;">aaaagg</td>
<td style="text-align: left;">agatct</td>
<td style="text-align: left;">ttatttgtatagttcatccatgcc</td>
</tr>
<tr class="odd">
<td>HIS3-ApaI-FW</td>
<td style="text-align: left;">aaaacc</td>
<td style="text-align: left;">gggccc</td>
<td style="text-align: left;">caaagATGacagagcagaaagc</td>
</tr>
<tr class="even">
<td>HIS3-XhoI-RV</td>
<td style="text-align: left;">aaaagg</td>
<td style="text-align: left;">ctcgag</td>
<td style="text-align: left;">ctacataagaacacctttggtgg</td>
</tr>
</tbody>
</table>
</section>
<section id="yeast-liquid-culture" class="level3">
<h3 class="anchored" data-anchor-id="yeast-liquid-culture">Yeast Liquid Culture</h3>
<section id="ypad" class="level4">
<h4 class="anchored" data-anchor-id="ypad">YPAD</h4>
<p>Yeast are easily grown in complex liquid media containing Yeast Extract, Peptone, Adenine, and Dextrose (D-Glucose), commonly abbreviated YPAD. 1x YPAD contains 1% (w/v) Bacto yeast extract, 2% (w/v) Bacto peptone. 2% (w/v) Glucose, and adenine hemisulfate 80 mg/l. For 1l of media, this corresponds to 10g yeast extract, 20g peptone, 20g Glucose, and 80mg adenine.</p>
</section>
<section id="synthetic-dropout-media" class="level4">
<h4 class="anchored" data-anchor-id="synthetic-dropout-media">Synthetic Dropout Media</h4>
<p>I am using Yeast Nitrogen Base and Complete Supplement Mixtures (CSM) from <a href="https://formedium.com/complete-supplement-mixture-csm-drop-outs/">Formedium</a>. The recipe for the YNB is as follows:</p>
<p>Formedium Yeast Nitrogen Base</p>
<div class="table-container" style="width: 50%; margin: auto;">
<table class="caption-top table">
<thead>
<tr class="header">
<th>Formula</th>
<th>mg/L</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Biotin</td>
<td>0.002</td>
</tr>
<tr class="even">
<td>Ca-Panthotenate</td>
<td>0.4</td>
</tr>
<tr class="odd">
<td>Folic acid</td>
<td>0.002</td>
</tr>
<tr class="even">
<td>Inositol</td>
<td>2</td>
</tr>
<tr class="odd">
<td>Nicotinic Acid (Niacin)</td>
<td>0.4</td>
</tr>
<tr class="even">
<td>p-Aminobenzoic Acid</td>
<td>0.2</td>
</tr>
<tr class="odd">
<td>Pyridoxine HCl</td>
<td>0.4</td>
</tr>
<tr class="even">
<td>Riboflavin</td>
<td>0.2</td>
</tr>
<tr class="odd">
<td>Thiamine HCl</td>
<td>0.4</td>
</tr>
<tr class="even">
<td>Boric Acid</td>
<td>0.5</td>
</tr>
<tr class="odd">
<td>Copper Sulfate</td>
<td>0.04</td>
</tr>
<tr class="even">
<td>Potassium Iodide</td>
<td>0.1</td>
</tr>
<tr class="odd">
<td>Ferric Chloride</td>
<td>0.2</td>
</tr>
<tr class="even">
<td>Manganese Sulfate</td>
<td>0.4</td>
</tr>
<tr class="odd">
<td>Sodium Molybdate</td>
<td>0.2</td>
</tr>
<tr class="even">
<td>Zinc Sulfate</td>
<td>0.4</td>
</tr>
<tr class="odd">
<td>Potassium Phosphate, monobasic</td>
<td>1000</td>
</tr>
<tr class="even">
<td>Magnesium Sulphate, anhydrous</td>
<td>500</td>
</tr>
<tr class="odd">
<td>Sodium Chloride</td>
<td>100</td>
</tr>
<tr class="even">
<td>Calcium Chloride, anhydrous</td>
<td>100</td>
</tr>
<tr class="odd">
<td>Ammonium Sulphate</td>
<td>5000</td>
</tr>
</tbody>
</table>
</div>
<p>Suspend <strong>6.9 g</strong> powdered medium in <strong>1 L</strong> distilled water.<br>
Store dry at room temperature.</p>
<p>The recipe per liter for the YNB from Sigma calls for 6.7g/l and in their work, Braun et al.&nbsp;used 1.7g/l for their plates and liquid media. Thus, we should use (1.7/6.7)*6.9=1.75g/l in our recreated recipe. Braun et al added 5g/l Ammonium Sulfate which corresponds to our YNB.</p>
<p>For the Complete supplement mixtures, Braun et al.&nbsp;used Sigma <a href="https://www.sigmaaldrich.com/GB/en/product/sigma/y2001">Synthetic dropout medium supplements Y2001</a> with all standard amino acids except for histidine, leucine, tryptophan and uracil. Each amino acid is present at a concentration of 76 mg/L. Other nutrients: Adenine (18 mg/L), inositol (76 mg/L), p-aminobenzoic acid (8 mg/L) and to which they added 0.006 g/liter L-tryptophan, and 0.003 g/liter uracil. Braun et al used the full 1.4g/l called for.</p>
<div class="table-container" style="width: 50%; margin: auto;">
<table class="caption-top table">
<thead>
<tr class="header">
<th>Formula</th>
<th>mg/L</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Adenine</td>
<td>10</td>
</tr>
<tr class="even">
<td>L-Arginine</td>
<td>50</td>
</tr>
<tr class="odd">
<td>L-Aspartic acid</td>
<td>80</td>
</tr>
<tr class="even">
<td>L-Histidine HCl</td>
<td>20</td>
</tr>
<tr class="odd">
<td>L-Isoleucine</td>
<td>50</td>
</tr>
<tr class="even">
<td>L-Leucine</td>
<td>100</td>
</tr>
<tr class="odd">
<td>L-Lysine HCl</td>
<td>50</td>
</tr>
<tr class="even">
<td>L-Methionine</td>
<td>20</td>
</tr>
<tr class="odd">
<td>L-Phenylalanine</td>
<td>50</td>
</tr>
<tr class="even">
<td>L-Threonine</td>
<td>100</td>
</tr>
<tr class="odd">
<td>L-Tryptophan</td>
<td>50</td>
</tr>
<tr class="even">
<td>L-Tyrosine</td>
<td>50</td>
</tr>
<tr class="odd">
<td>Uracil</td>
<td>20</td>
</tr>
<tr class="even">
<td>Valine</td>
<td>140</td>
</tr>
<tr class="odd">
<td><strong>TOTAL</strong></td>
<td><strong>790</strong></td>
</tr>
</tbody>
</table>
</div>
</section>
</section>
<section id="yeast-plating" class="level3">
<h3 class="anchored" data-anchor-id="yeast-plating">Yeast Plating</h3>
<p>Liquid cultures of yeast are diluted and spread on solid media plates in order to facilitate the selection of a colony, which is presumably generated from a single cell. This procedure <em>minimizes</em> genetic variation in the population of cells used to start a new culture, but also bottlenecks the culture (potentially fixing any mutations that have arisen). For this procedure to work, the liquid culture must be sufficiently diluted so that only <span class="math inline">\(\approx10^1-10^2\)</span> cells are spread on each plate. The following will describe how to carry out this procedure.</p>
<section id="producing-solid-media-plates" class="level4">
<h4 class="anchored" data-anchor-id="producing-solid-media-plates">Producing solid media plates</h4>
<p>Generally the solid media used is a solidified gel made of <a href="https://en.wikipedia.org/wiki/Agar">Agar</a> derived from red algae. For general growth and maintenance it is most important that the yeast have sufficient nutrients, and the exact composition is generally less important, so for these tasks, so-called <em>complex</em> media is used, the standard being YPD (Yeast extract, Peptone, Dextrose) media. To make YPD Agar plates, the yeast extract, peptone, and agar are autoclaved together, then cooled to around 65C and the sterile glucose is added aseptically. The liquified agar media is then poured into dishes (Petri dishes) of the appropriate size (90 mm).</p>
</section>
<section id="dextroseglucose-solution" class="level4">
<h4 class="anchored" data-anchor-id="dextroseglucose-solution">Dextrose/Glucose Solution</h4>
<p>To start measure 100ml of <a href="https://en.wikipedia.org/wiki/Purified_water#Distillation">purified</a> water using a <a href="https://en.wikipedia.org/wiki/Graduated_cylinder">graduated cylinder</a>. Then measure 40g of glucose. If you have a hot-plate stirrer, set the hot plate to 75C and place a magnetic stir bar into a 250ml <a href="https://en.wikipedia.org/wiki/Erlenmeyer_flask">Erlenmeyer flask</a>. Slowly add the glucose to the stirred water about 5 grams at a time, allowing it to completely dissolve between additions. The glucose solution is not autoclaved, as it can lead to degradation. Therefore it is necessary to sterilize it by passing it through a filter with 0.22 micron pore size (“sterile filtering”) which is small enough to remove most bacteria and viruses.</p>
</section>
<section id="selection-plates" class="level4">
<h4 class="anchored" data-anchor-id="selection-plates">Selection plates</h4>
<p>For this work, selection plates were made from yeast nitrogen base without amino acids from <a href="https://formedium.com/product/yeast-nitrogen-base-without-amino-acids/">Formedium</a> was used. To this <a href="https://formedium.com/product/csm-single-drop-outs/">CSM single drop outs</a> for leucine, histidine, or both were added (both containing additional adenosine to 40mg/l). These are prepared in the same way as the complex media plates, with the addition of glucose or galactose done aseptically after autoclaving.</p>
</section>
</section>
<section id="yeast-strains-1" class="level3">
<h3 class="anchored" data-anchor-id="yeast-strains-1">Yeast Strains</h3>
<p>A yeast strain is a term used to indicate a yeast that has a collection of defined mutations relative to a reference strain, often called the <em>wild-type</em> strain. Sometimes strains are created to have properties that make them easier to work with, for example, a strain might carry a mutation that renders it unable to grow without an externally provided amino acid. When growing this strain, the growth can be controlled by limiting how much of this amino acid is provided. A list of commonly used yeast strains can be found on the <a href="https://sites.google.com/view/yeastgenome-help/more-about-yeast/commonly-used-strains">Saccharomyces Genome Database</a></p>
<section id="w303" class="level5">
<h5 class="anchored" data-anchor-id="w303">W303</h5>
<p>We are working with a derivative of the <a href="https://www.yeastgenome.org/strain/w303">W303</a> strain, which is itself a derivative of the <a href="https://www.yeastgenome.org/strain/S288C">S88C</a> strain. This strain has genotype MATa/MATα <em>{leu2-3,112 trp1-1 can1-100 ura3-1 ade2-1 his3-11,15} [phi+]</em></p>
</section>
</section>
<section id="transformation-1" class="level3">
<h3 class="anchored" data-anchor-id="transformation-1">Transformation</h3>
<p>Transformation refers to the uptake of external or foreign DNA into a cell. This is usually achieved by disrupting the membrane in some way, such as using chemicals (chemical transformation), large electrical fields (electoporation), or high temperatures (heat shock transformation). Transformation generally involves making <a href="#competent-cells">Competent cells</a> which are cells prepared in such a way that they more easily take up foreign DNA, and then introducing that foreign DNA into the cells by either chemical, heat-shock, or electroporation. For yeast, a combination of the chemical <a href="#lithium-acetate-transformation">Lithium Acetate</a> and heat-shock is often used.</p>
<section id="competent-cells" class="level4">
<h4 class="anchored" data-anchor-id="competent-cells">Competent Cells</h4>
<p>Competent cells are cells at are specially prepared to be receptive to the uptake of external DNA. For most transformation procedures, this involves starting with a healthy exponentially growing population, concentrating it, and in the case of chemical or thermal transformation, adding a chemical that aids in membrane permeabilization, such as dimethyl sulfoxide (DMSO). FInally, if the cells are to be frozen for later use, a cryoprotectant such as glycerol is added. We made competent cells following the protocol in <span class="citation" data-cites="gietzFrozenCompetentYeast2007">&nbsp;[<a href="#ref-gietzFrozenCompetentYeast2007" role="doc-biblioref">8</a>]</span>.</p>
</section>
<section id="lithium-acetate-transformation" class="level4">
<h4 class="anchored" data-anchor-id="lithium-acetate-transformation">Lithium Acetate Transformation</h4>
<p>The LiAc (lithium acetate) protocol <span class="citation" data-cites="itoTransformationIntactYeast1983">&nbsp;[<a href="#ref-itoTransformationIntactYeast1983" role="doc-biblioref">7</a>]</span> is a popular method for yeast transformation that is a combination of chemical and heat shock transformations. Competent yeast cells are treated with lithium acetate, which permeabilizes the cell wall. Additional single stranded DNA, often boiled salmon sperm DNA (boiling denatures the DNA making it single stranded), and polyethylene glycol (PEG) are added to facilitate DNA uptake, and a brief heat shock helps the DNA enter the cells. We used the protocols outlines in <span class="citation" data-cites="gietzHighefficiencyYeastTransformation2007">&nbsp;[<a href="#ref-gietzHighefficiencyYeastTransformation2007" role="doc-biblioref">9</a>]</span>. A quicker method for easier transformations is also given in <span class="citation" data-cites="gietzQuickEasyYeast2007">&nbsp;[<a href="#ref-gietzQuickEasyYeast2007" role="doc-biblioref">10</a>]</span>,</p>
</section>
</section>
<section id="selectable-markers" class="level3">
<h3 class="anchored" data-anchor-id="selectable-markers">Selectable Markers</h3>
<p>Selectable markers are a combination of a selection pressure, and a gene allows a microbe to avoid that selective pressure. One of the most commonly used selectable markers is the combination of an antibiotic and an antibiotic resistance gene in bacteria. In this scheme, plasmids to be transformed are engineered to contain an antibiotic resistance gene, such as beta-lactamase, and then transformed bacteria are grown on plates that contain a beta-lactam antibiotic, such as ampicillin. There are many bacterial antibiotic/resistance markers, such as chloramphenicol, kanamycin, and tetracycline. Continued growth on antibiotic containing plates even after the initial transformation allows one to ensure that the plasmid is maintained in the population. Growth on non-selective media can result in plasmid loss over time. In yeast, most commonly auxotrophic markers are used. In this case, the selective pressure is a deletion of a native gene used to make a particular metabolite, often an amino acid. These cells then require that amino acid to be supplemented in the growth media. However, if the cell is transformed with the gene or an enzyme that rescues its ability to make this metabolite, it can then be grown in media without that supplemental metabolite. In this way, continued growth in a media without the metabolite ensures that the plasmid is maintained.</p>



</section>
</section>
</section>


<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body" data-entry-spacing="0" role="list">
<div id="ref-Stolovicki2006-sd" class="csl-entry" role="listitem">
<div class="csl-left-margin">[1] </div><div class="csl-right-inline">E. Stolovicki, T. Dror, N. Brenner, and E. Braun, <em>Synthetic Gene Recruitment Reveals Adaptive Reprogramming of Gene Regulation in Yeast</em>, Genetics <strong>173</strong>, 75 (2006).</div>
</div>
<div id="ref-woronoffMetabolicCostRapid2020" class="csl-entry" role="listitem">
<div class="csl-left-margin">[2] </div><div class="csl-right-inline">G. Woronoff, P. Nghe, J. Baudry, L. Boitard, E. Braun, A. D. Griffiths, and J. Bibette, <em><a href="https://doi.org/10.1073/pnas.1913767117">Metabolic Cost of Rapid Adaptation of Single Yeast Cells</a></em>, Proc. Natl. Acad. Sci. U.S.A. <strong>117</strong>, 10660 (2020).</div>
</div>
<div id="ref-cormackYeastenhancedGreenFluorescent1997" class="csl-entry" role="listitem">
<div class="csl-left-margin">[3] </div><div class="csl-right-inline">B. P. Cormack, G. Bertram, M. Egerton, N. A. R. Gow, S. Falkow, and A. J. P. Brown, <em><a href="https://doi.org/10.1099/00221287-143-2-303">Yeast-Enhanced Green Fluorescent Protein (<span class="nocase">yEGFP</span>): A Reporter of Gene Expression in <span>Candida</span> Albicans</a></em>, Microbiology <strong>143</strong>, 303 (1997).</div>
</div>
<div id="ref-cormackFACSoptimizedMutantsGreen1996" class="csl-entry" role="listitem">
<div class="csl-left-margin">[4] </div><div class="csl-right-inline">B. P. Cormack, R. H. Valdivia, and S. Falkow, <em><a href="https://doi.org/10.1016/0378-1119(95)00685-0"><span class="nocase">FACS-optimized</span> Mutants of the Green Fluorescent Protein (<span>GFP</span>)</a></em>, Gene <strong>173</strong>, 33 (1996).</div>
</div>
<div id="ref-kaishimaExpressionVariedGFPs2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">[5] </div><div class="csl-right-inline">M. Kaishima, J. Ishii, T. Matsuno, N. Fukuda, and A. Kondo, <em><a href="https://doi.org/10.1038/srep35932">Expression of Varied <span>GFPs</span> in <span>Saccharomyces</span> Cerevisiae: Codon Optimization Yields Stronger Than Expected Expression and Fluorescence Intensity</a></em>, Sci Rep <strong>6</strong>, 35932 (2016).</div>
</div>
<div id="ref-bennettMetabolicGeneRegulation2008" class="csl-entry" role="listitem">
<div class="csl-left-margin">[6] </div><div class="csl-right-inline">M. R. Bennett, W. L. Pang, N. A. Ostroff, B. L. Baumgartner, S. Nayak, L. S. Tsimring, and J. Hasty, <em><a href="https://doi.org/10.1038/nature07211">Metabolic Gene Regulation in a Dynamically Changing Environment</a></em>, Nature <strong>454</strong>, 1119 (2008).</div>
</div>
<div id="ref-itoTransformationIntactYeast1983" class="csl-entry" role="listitem">
<div class="csl-left-margin">[7] </div><div class="csl-right-inline">H. Ito, Y. Fukuda, K. Murata, and A. Kimura, <em><a href="https://doi.org/10.1128/jb.153.1.163-168.1983">Transformation of Intact Yeast Cells Treated with Alkali Cations</a></em>, J Bacteriol <strong>153</strong>, 163 (1983).</div>
</div>
<div id="ref-gietzFrozenCompetentYeast2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">[8] </div><div class="csl-right-inline">R. D. Gietz and R. H. Schiestl, <em><a href="https://doi.org/10.1038/nprot.2007.17">Frozen Competent Yeast Cells That Can Be Transformed with High Efficiency Using the <span>LiAc</span>/<span>SS</span> Carrier <span>DNA</span>/<span>PEG</span> Method</a></em>, Nat Protoc <strong>2</strong>, 1 (2007).</div>
</div>
<div id="ref-gietzHighefficiencyYeastTransformation2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">[9] </div><div class="csl-right-inline">R. D. Gietz and R. H. Schiestl, <em><a href="https://doi.org/10.1038/nprot.2007.13">High-Efficiency Yeast Transformation Using the <span>LiAc</span>/<span>SS</span> Carrier <span>DNA</span>/<span>PEG</span> Method</a></em>, Nat Protoc <strong>2</strong>, 31 (2007).</div>
</div>
<div id="ref-gietzQuickEasyYeast2007" class="csl-entry" role="listitem">
<div class="csl-left-margin">[10] </div><div class="csl-right-inline">R. D. Gietz and R. H. Schiestl, <em><a href="https://doi.org/10.1038/nprot.2007.14">Quick and Easy Yeast Transformation Using the <span>LiAc</span>/<span>SS</span> Carrier <span>DNA</span>/<span>PEG</span> Method</a></em>, Nat Protoc <strong>2</strong>, 35 (2007).</div>
</div>
</div></section><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>enzymes that recognize and cut a specific DNA sequence<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/notes\.livingphysics\.org");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>