---
title: "Recreating the Gal-His yeast system used to demonstrate adaptive reprogramming"
author: 
 - name: David Jordan
   url: mailto:public@livingphysics.org
 - name: Mihoko Tame
code-fold: true
date: "2025-01-28"
categories: [Builds, Notes, Yeast]
bibliography: references.bib
csl: ../styles/american-physical-society.csl
draft: true
image: "images/Project_Image002_ch00.jpg"
feed:
    image-height: 60
    image-width: 60
---

> A goal for the future would be to determine the extent of
knowledge the cell has of itself and how it uses that knowledge
in a thoughtful manner when challenged.
> 
> (McClintock 1984)

This note describes how I am recreating the strain from the Braun lab [-@Stolovicki2006-sd] used to demonstrate adaptive reprogramming to novel challenges in the yeast *Saccharomyces cerevisiae*  Unfortunately the strain and plasmid were lost due to a freezer fault. This document is written informally but should be understandable to a reasonably interested secondary school student.  For brevity, the main document uses standard scientific terminology, but these terms are explained in detail in the [Procedures](#procedures) section.

# Overview

## Plasmid Reconstruction
The original plasmid seems to have been generated by [cloning](#cloning) the His3 gene and a GFP gene into the dual expression plasmid [pESC-Leu](https://www.agilent.com/en/product/protein-expression/protein-expression-vectors-kits/yeast-expression-vectors/pesc-yeast-epitope-tagging-vectors-232966) from Agilent.  It is unclear what GFP gene was used in the original plasmid, but one could use a mutationally optimized GFP originally introduced in [-@cormackYeastenhancedGreenFluorescent1997].  Further information on optimized GFPs can be found in [-@cormackFACSoptimizedMutantsGreen1996], and a comparison of yeast optimized GFPs can be found in [-@kaishimaExpressionVariedGFPs2016].  The plasmid has a Leucine selectable marker and a $2\mu$ origin for yeast a and AmpR selective marker and $ori$ origin for growth in *E. coli*.  The plasmid contains a bidirectional Gal1/Gal10 promoter and each promoter has a multiple cloning site (MCS). 

### Cloning
We have used the GFP cloned from the [pKEN GFP mut2](https://www.addgene.org/20409/) plasmid. The HIS3 gene was cloned from the plasmid [pRS33](https://www.snapgene.com/plasmids/yeast_plasmids/pRS313).  MT designed forward and reverse PCR primers to [introduce restriction sites](#restriction-pcr) for NotI and BglII into the amplified GFP sequence and ApaI and XhoI in the HIS3 gene.  The GFP sequence was cloned into the pESC-Leu Gal10 MCS and the HIS3 into the Gal1 MCS in the bi-directional Gal1/Gal10 promoter.  Cloning results were confirmed by whole plasmid sequencing.  

## Yeast Strains
It would be very simple to use store bought yeast for these experiments, and I plan to do that for simpler experiments where no genetic engineering is necessary, however, laboratory strains have many features that make genetic engineering muh easier, such as gene deletions that can be used as [selectable markers](#selectable-markers).  For these experiments, we need a yeast strain that cannot make its own Leucine, so that we can use the Lu selectable marker on the plasmid, as well as being unable to make histidine, so that Gal-His construct will be its sole source of Histidine during adaptation trials. 

The original experiments were done in a yeast strain designated as YPH499. Interestingly, in the supplemental information of this paper by [-@bennettMetabolicGeneRegulation2008], the authors note that this strain has "significantly impaired galactose uptake".  This strain is derived from S288C, which has a known defect in the GAL2 galactose permease gene (a transporter that brings galactose into the cell).  The *GAL2* gene was supposed to have been repaired in YPH499 but the authors of the above note that there remain a number of point mutations in the coding sequence of the Gal2 galactose transporter compared to the strain K699. They provide evidence that galactose uptake is impaired in the YPH499 strain.  We are using the W303 strain, which seems to have the same Gal2 sequence as the K699 stain save one residue at 369. 

>Our sequencing results revealed nine point mutations in the YPH499 *GAL2* sequence, yielding the following five amino acid mutations: V8M P50S S90G Y369S R392H.

They then use a fluorescent fusion GAL2p and show that is distribution is not uniform on the cell membrane in YPH499 as it is in another strain K699 which has teh *wild-type* *GAL* sequence.   The strains I have easily available are the W303 strain and the BY4743 strains. The W303 strain is only 85.4% congenic with S288c (and thus with YPH499) 

For Gal2, w303 is closer to K699 than to YPH499. For the residues noted, K699 has the residues [VPSYR], YPH499 has the residues [MSGSH] and our strain W303 has [VPSSR] according to publicly available [sequencing data](https://www.yeastgenome.org/variant-viewer#/S000004071).

# Procedures
## Cloning
### Generating the DNA inserts
This usually involves either cutting the DNA out of an existing source using restriction enzymes[^1], having the DNA synthesized, or most commonly, amplifying it out of an existing source using the Polymerase Chain Reaction (PCR). 

#### PCR amplification

### Plasmids
Plasmids are circles of DNA containing a sequence that tells a certain cell type to replicate it.  These are known and origins of replication.  Often plasmids have origins that alow them to be replicated in bacteria, as transforming bacteria and growing them is convenient way to *amplify* the plasmid. For example, the [pESC-Leu](https://www.snapgene.com/plasmids/yeast_plasmids/pESC-LEU) plasmid that we hav used as a backbone for this work contains three origins shown in yellow in the the plasmid map.  The bacterial double stranded DNA origin is denote *ori*.  The *f1 ori* is a phage derived origin that allows for single stranded replication and packaging into a phage. Both of these operate in bacteria.  The $2\mu$ origin is the yeast replication origin.  The identity of the origin control for example the plasmid copy number in the cell. In addition, cells usually cannot maintain reliably different plasmids with the same origin. 

![pESC-Leu PLasmid Map (from SnapGene)](./images/pESC-LEU.png)

For more information about plasmid origins of replication, see [this Addgene article](https://blog.addgene.org/plasmid-101-origin-of-replication)

### Restriction Enzymes
Restriction enzymes are proteins that recognize a specific sequence in DNA and cleave hte DNA in a predictable manner. 

### Restriction PCR
Resriction sites 
### Yeast Plating
Liquid cultures of yeast are diluted and spread on solid media plates in order to facilitate the selection of a colony, which is presumably generated from a single cell.  This procedure *minimizes* genetic variation in the population of cells used to start a new culture, but also bottlenecks the culture (potentially fixing any mutations that have arisen).  For this procedure to work, the liquid culture must be sufficiently diluted so that only $\approx10^1-10^2$ cells are spread on each plate.  The following will describe how to carry out this procedure.

#### Producing solid media plates
Generally the solid media used is a solidified gel made of [Agar](https://en.wikipedia.org/wiki/Agar) derived from red algae.  For general growth and maintenance it is most important that the yeast have sufficient nutrients, and the exact composition is generally less important, so for these tasks, so-called *complex* media is used, the standard being YPD (Yest extract, Peptone, Dextrose) media. To make YPD Agar plates, the yeast extract, peptone, and agar are autoclaved together, then cooled to around 65C and the sterile glucose is added aseptically.  The liquified agar media is then poured into dishes (Petri dishes) of the appropriate size (90 mm).  

#### Dextrose/Glucose Solution
To start measure 100ml of [purified](https://en.wikipedia.org/wiki/Purified_water#Distillation) water using a [graduated cylinder](https://en.wikipedia.org/wiki/Graduated_cylinder).  Then measure 40g of glucose.  If you have a hot-plate stirrer, set the hot plate to 75C and place a magnetic stir bar into a 250ml [Erlenmeyer flask](https://en.wikipedia.org/wiki/Erlenmeyer_flask).  Slowly add the glucose to the stirred water about 5 grams at a time, allowing it to completely dissolve between additions. The glucose solution is not autoclaved, as it can lead to degradation.  Therefore it is necessary to sterilize it by passing it through a filter with 0.22 micron pore size ("sterile filtering") which is small enough to remove most bacteria and viruses.  

#### Yeast Strains
A yeast strain is a term used to indicate a yest that has a collection of defined mutations relative to a reference strain, often called the *wild-type* strain.  Sometimes strains are created to have properties that make them easier to work with, for example, a strain might carry a mutation that renders it unable to grow without an externally provided amino acid.  When growing this strain, the growth can be controlled by limiting how much of this amino acid is provided. A list of commonly used yeast strains can be found on the [Saccharomyces Genome Database](https://sites.google.com/view/yeastgenome-help/more-about-yeast/commonly-used-strains)
##### W303 
We are working with a derivative of the [W303](https://www.yeastgenome.org/strain/w303) strain, which is itself a derivative of the [S88C](https://www.yeastgenome.org/strain/S288C) strain. This strain has genotype MATa/MATÎ± *{leu2-3,112 trp1-1 can1-100 ura3-1 ade2-1 his3-11,15} [phi+]*

### Transformation
Lorem ipsum

### Selectable Markers
Lorem ipsum

[^1]: enzymes that recognize and cut a specific DNA sequence